### ISSUE TITLE
no-shadow potential false positive

### ISSUE THREAD

**Issue by myagoo** [eslint/eslint#6574]
**What version of ESLint are you using?**
2.4.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```json
{
    "extends": ["airbnb/base", "angular", "plugin:lodash/recommended"],
    "env": {
        "browser": true,
        "node": true,
        "es6": true
    },
    "globals": {
        "_": false,
        "moment": false,
        "inject": false,
        "d3": false,
        "Highcharts": false,
        "HighchartsMore": false,
    },
    "rules": {
        "comma-dangle": [2, "never"],
        "curly": [2, "all"],
        "indent": [2, 4],
        "angular/definedundefined": 0,
        "angular/json-functions": 0,
        "angular/no-private-call": 0,
        "angular/no-service-method": 0,
        "angular/log": 0,
        "angular/no-digest": 0,
        "spaced-comment": 1,
        "max-len": 0,
        "no-param-reassign": 0,
        "arrow-body-style": 0,
        "prefer-template": 0,
        "consistent-return": 0,
        "quote-props": 0,
        "no-return-assign": 0,
        "prefer-rest-params": 0,
        "prefer-const": 0,
        "array-callback-return": 0,
        "no-unneeded-ternary": 0,
        "no-useless-constructor": 0,
        "quotes": 0,
        "no-case-declarations": 0,
        "no-else-return": 0,
        "lodash/prefer-lodash-method": 0,
        "lodash/matches-shorthand": [2, "never"],
        "lodash/matches-prop-shorthand": [2, "never"],
        "lodash/prefer-constant": 0,
        "lodash/path-style": 0
    },
    "plugins": [
        "angular",
        "lodash"
    ]
}
```
**What did you do? Please include the actual source code causing the issue.**
```javascript
const group = _.find(vm.groups, (group) => group.id === key);
```

**What did you expect to happen?**
To not have a warning

**What actually happened? Please include the actual, raw output from ESLint.**
`error  'group' is already declared in the upper scope  no-shadow`

**Comment by ilyavolodin** [comment]
You have two declarations of `group` variable:

```js
const group = ...
```

and

```js
(group) => group.id === key
```

It looks like the rule is correctly reporting that you are declaring an argument inside arrow function that would shadow top level `group` variable.
